src/HtmlQuotations.ts

```diff
@@ -112,33 +111,41 @@ export function deleteQuotationTags(document: Document, element: Node, quotation
     isTagInQuotation
   };
 }
+export interface cutQuoteOption {
+  onlyRemoveEmptyBlocks?: boolean
+}
 
 /**
  * Cuts the outermost block element with the class "gmail_quote".
  *
  * @param {Document} document - The document to cut the element from.
+ * @param {cutQuoteOption} options - Extra options.
  * @return {boolean} Whether a corresponding quote was found or not.
  */
-export function cutGmailQuote(document: Document): boolean {
+export function cutGmailQuote(document: Document, options?: cutQuoteOption): boolean {
   // Find the first element that fits our criteria.
   const gmailQuote = <Node>XPath.select("//*[contains(@class, 'gmail_quote')]", document, true);
 
   // If no quote was found, or if that quote was a forward, return false.
-  if (!gmailQuote || (gmailQuote.textContent && matchStart(gmailQuote.textContent, ForwardRegexp)))
+  if (!gmailQuote || (gmailQuote.textContent && matchStart(gmailQuote.textContent, ForwardRegexp)) || shouldNotRemoveQuote(gmailQuote, options))
     return false;
 
   // Otherwise, remove the quote from the document and return.
   gmailQuote.parentNode.removeChild(gmailQuote);
   return true;
 };
 
+function shouldNotRemoveQuote(node: Node, options?: cutQuoteOption) {
```

Comment:
This is used by functions both above and below.
I feel like it should be either at the beginning or at the end.